contrib/dracut/90zfs/mount-zfs.sh.in

```diff
@@ -10,17 +10,29 @@ case "${root}" in
 	*) return ;;
 esac
 
-if command -v systemctl >/dev/null; then
-	# If sysroot.mount exists, the initial RAM disk configured
-	# it to mount ZFS on root.  In that case, we bail early.
-	loadstate="$(systemctl --system --show -p LoadState sysroot.mount || true)"
-	if [ "${loadstate}" = "LoadState=not-found" -o "${loadstate}" = "" ] ; then
-		info "ZFS: sysroot.mount absent, mounting root with mount-zfs.sh"
-	else
-		info "ZFS: sysroot.mount present, delegating root mount to it"
-		return
-	fi
+GENERATOR_FILE=/run/systemd/generator/sysroot.mount
+GENERATOR_EXTENSION=/run/systemd/generator/sysroot.mount.d/zfs-enhancement.conf
+
+if [ -e "$GENERATOR_FILE" -a -e "$GENERATOR_EXTENSION" ] ; then
```

Comment:
Ok, so you're now using the existence of these two files to indicate that the `sysroot.mount` method should be used.